transforms/angle-brackets/transform.js

```diff
@@ -228,6 +228,8 @@ function transformToAngleBracket(env, fileInfo, config) {
       firstParamOutput = b.attr('@query', b.mustache(b.path('hash'), [], firstParamInput.hash));
     } else if (firstParamInput.type === 'PathExpression') {
       firstParamOutput = b.attr('@route', b.mustache(firstParamInput.original));
+    } else if (firstParamInput.type === 'SubExpression') {
+      firstParamOutput = b.attr('@route', b.mustache(firstParamInput.path, firstParamInput.params));
```

Comment:
Also need to pass `firstParamInput.hash`. Can you add another test case for this:

Input:

```hbs
{{#link-to (random-helper someVariable ".detail" do-the="thing") class="some-class"}}click{{/link-to}}
```

Output:

```hbs
<LinkTo @route={{random-helper someVariable ".detail" do-the="thing"}} class="some-class">click</LinkTo>
```

Reply:
Updated the PR with your test + fix